.github/workflows/test.yml

```diff
@@ -1,22 +1,57 @@
-name: test
+name: CI & Release
+
 on:
-  - push
-  - pull_request
+  push:
+  pull_request:
+    branches:
+      - alpha
+      - beta
+      - main
+
 jobs:
   test:
-    name: Node.js ${{ matrix.node-version }}
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.platform }}
+    name: Node.js ${{ matrix.node-version }} / ${{ matrix.platform }}
     strategy:
       fail-fast: false
       matrix:
-        node-version:
-          - 16
-          - 14
-          - 12
+        platform: [ubuntu-latest, macos-latest]
```

Comment:
I've seen us use this pattern in other repos and in practice we have far fewer `macos-latest` runners available and it frequently results in long wait times when you have multiple PRs on multiple repos that need merging. I recommend following the pattern we use on repos like sanity client instead, where you have exactly one run of `macos-latest and `windows-latest` and have the rest be strictly `ubuntu-latest` 🤔

Reply:
To be honest, I think it's (almost) pointless to run tests on windows/mac for modules that do not even use node APIs. There's bound to be differences and pitfalls when doing things like command spawning, file system operations and such - but for pure JS modules it's hard to see where it could fail cross-platform.